test/coverage/communication/PublisherTest.cpp

```diff
@@ -162,4 +165,36 @@ TEST_F(TestPublisher, PublishSuccessWithoutPriority) {
 	          UPriority::UPRIORITY_CS1);
 }
 
+TEST_F(TestPublisher, PubSubSuccess) {
+	// sub
+	auto transportSub =
+	    std::make_shared<uprotocol::test::UTransportMock>(source_);
+
+	auto callback = [this](auto arg1) {
+		return this->handleCallbackMessage(arg1);
+	};
+
+	auto result =
+	    Subscriber::subscribe(transportSub, topic_, std::move(callback));
+
+	// pub
+	std::string testPayloadStr = "test_payload";
+	Publisher publisher(transportMock_, std::move(topic_), format_, priority_,
+	                    ttl_);
+
+	uprotocol::v1::UStatus retval;
+	retval.set_code(uprotocol::v1::UCode::OK);
+	transportMock_->send_status_ = retval;
+
+	Payload testPayload(testPayloadStr, format_);
+	auto status = publisher.publish(std::move(testPayload));
+
+	// // Manually bridge the two transports
+	transportSub->mockMessage(transportMock_->message_);
```

Comment:
Add a step before this confirming the source/sink of the published message match the source/sink filters of the publisher's listener registration.

Reply:
done